File: frontend/src/lib/api/organization.ts
import { apiClient } from './index';

// ============================================================================
// TYPES
// ============================================================================

export interface Team {
  id: string;
  org_id: string;
  name: string;
  description?: string;
  team_lead_id?: string;
  parent_team_id?: string;
  status: string;
  metadata: string;
  created_at: string;
  updated_at: string;
  created_by?: string;
  team_lead?: {
    id: string;
    full_name: string;
    email: string;
    username: string;
  };
  members?: TeamMember[];
  member_count: number;
}

export interface TeamMember {
  id: string;
  team_id: string;
  user_id: string;
  role: string;
  joined_at: string;
  is_active: boolean;
  full_name: string;
  email: string;
  username: string;
}

export interface Project {
  id: string;
  org_id: string;
  name: string;
  description?: string;
  project_manager_id?: string;
  status: string;
  priority: string;
  start_date?: string;
  end_date?: string;
  budget?: number;
  progress: number;
  metadata: string;
  created_at: string;
  updated_at: string;
  created_by?: string;
  project_manager?: {
    id: string;
    full_name: string;
    email: string;
    username: string;
  };
  teams?: ProjectTeam[];
  members?: ProjectMember[];
  team_count: number;
  member_count: number;
}

export interface ProjectTeam {
  id: string;
  project_id: string;
  team_id: string;
  assigned_at: string;
  team_name: string;
  team_member_count: number;
}

export interface ProjectMember {
  id: string;
  project_id: string;
  user_id: string;
  role: string;
  allocation_percentage: number;
  joined_at: string;
  is_active: boolean;
  full_name: string;
  email: string;
  username: string;
}

export interface Group {
  id: string;
  org_id: string;
  name: string;
  description?: string;
  group_type: string;
  owner_id?: string;
  status: string;
  metadata: string;
  created_at: string;
  updated_at: string;
  created_by?: string;
  owner?: {
    id: string;
    full_name: string;
    email: string;
    username: string;
  };
  members?: GroupMember[];
  member_count: number;
}

export interface GroupMember {
  id: string;
  group_id: string;
  user_id: string;
  role: string;
  joined_at: string;
  is_active: boolean;
  full_name: string;
  email: string;
  username: string;
  team_name?: string;
}

export interface Workspace {
  id: string;
  org_id: string;
  name: string;
  description?: string;
  workspace_type: string;
  team_id?: string;
  project_id?: string;
  owner_id?: string;
  settings: string;
  is_private: boolean;
  created_at: string;
  updated_at: string;
}

// ============================================================================
// TEAMS API
// ============================================================================

export const teamsAPI = {
  create: (orgId: string, data: {
    name: string;
    description?: string;
    team_lead_id?: string;
    parent_team_id?: string;
  }) => apiClient.post(`/api/v1/organizations/${orgId}/teams`, data),

  list: (orgId: string, params?: { status?: string }) =>
    apiClient.get(`/api/v1/organizations/${orgId}/teams`, { params }),

  get: (teamId: string) =>
    apiClient.get(`/api/v1/teams/${teamId}`),

  update: (teamId: string, data: {
    name?: string;
    description?: string;
    team_lead_id?: string;
    status?: string;
  }) => apiClient.put(`/api/v1/teams/${teamId}`, data),

  delete: (teamId: string) =>
    apiClient.delete(`/api/v1/teams/${teamId}`),

  // Team Members
  addMember: (teamId: string, data: {
    user_id: string;
    role?: string;
  }) => apiClient.post(`/api/v1/teams/${teamId}/members`, data),

  removeMember: (teamId: string, userId: string) =>
    apiClient.delete(`/api/v1/teams/${teamId}/members/${userId}`),

  listMembers: (teamId: string) =>
    apiClient.get(`/api/v1/teams/${teamId}/members`),
};

// ============================================================================
// PROJECTS API
// ============================================================================

export const projectsAPI = {
  create: (orgId: string, data: {
    name: string;
    description?: string;
    project_manager_id?: string;
    priority?: string;
    start_date?: string;
    end_date?: string;
    budget?: number;
  }) => apiClient.post(`/api/v1/organizations/${orgId}/projects`, data),

  list: (orgId: string, params?: {
    status?: string;
    priority?: string;
  }) => apiClient.get(`/api/v1/organizations/${orgId}/projects`, { params }),

  get: (projectId: string) =>
    apiClient.get(`/api/v1/projects/${projectId}`),

  update: (projectId: string, data: {
    name?: string;
    description?: string;
    status?: string;
    priority?: string;
    progress?: number;
    budget?: number;
  }) => apiClient.put(`/api/v1/projects/${projectId}`, data),

  delete: (projectId: string) =>
    apiClient.delete(`/api/v1/projects/${projectId}`),

  // Project Teams
  assignTeam: (projectId: string, data: {
    team_id: string;
  }) => apiClient.post(`/api/v1/projects/${projectId}/teams`, data),

  removeTeam: (projectId: string, teamId: string) =>
    apiClient.delete(`/api/v1/projects/${projectId}/teams/${teamId}`),

  // Project Members
  addMember: (projectId: string, data: {
    user_id: string;
    role?: string;
    allocation_percentage?: number;
  }) => apiClient.post(`/api/v1/projects/${projectId}/members`, data),

  removeMember: (projectId: string, userId: string) =>
    apiClient.delete(`/api/v1/projects/${projectId}/members/${userId}`),
};

// ============================================================================
// GROUPS API
// ============================================================================

export const groupsAPI = {
  create: (orgId: string, data: {
    name: string;
    description?: string;
    group_type?: string;
    owner_id?: string;
  }) => apiClient.post(`/api/v1/organizations/${orgId}/groups`, data),

  list: (orgId: string, params?: { group_type?: string }) =>
    apiClient.get(`/api/v1/organizations/${orgId}/groups`, { params }),

  get: (groupId: string) =>
    apiClient.get(`/api/v1/groups/${groupId}`),

  update: (groupId: string, data: {
    name?: string;
    description?: string;
    status?: string;
  }) => apiClient.put(`/api/v1/groups/${groupId}`, data),

  delete: (groupId: string) =>
    apiClient.delete(`/api/v1/groups/${groupId}`),

  // Group Members
  addMember: (groupId: string, data: {
    user_id: string;
    role?: string;
  }) => apiClient.post(`/api/v1/groups/${groupId}/members`, data),

  removeMember: (groupId: string, userId: string) =>
    apiClient.delete(`/api/v1/groups/${groupId}/members/${userId}`),
};

// ============================================================================
// WORKSPACES API
// ============================================================================

export const workspacesAPI = {
  create: (orgId: string, data: {
    name: string;
    description?: string;
    workspace_type?: string;
    team_id?: string;
    project_id?: string;
    is_private?: boolean;
  }) => apiClient.post(`/api/v1/organizations/${orgId}/workspaces`, data),

  list: (orgId: string, params?: {
    team_id?: string;
    project_id?: string;
  }) => apiClient.get(`/api/v1/organizations/${orgId}/workspaces`, { params }),
};

File: services/org/models/organization.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Team struct {
	ID           uuid.UUID  `db:"id" json:"id"`
	OrgID        uuid.UUID  `db:"org_id" json:"org_id"`
	Name         string     `db:"name" json:"name"`
	Description  *string    `db:"description" json:"description,omitempty"`
	TeamLeadID   *uuid.UUID `db:"team_lead_id" json:"team_lead_id,omitempty"`
	ParentTeamID *uuid.UUID `db:"parent_team_id" json:"parent_team_id,omitempty"`
	Status       string     `db:"status" json:"status"`
	Metadata     string     `db:"metadata" json:"metadata"`
	CreatedAt    time.Time  `db:"created_at" json:"created_at"`
	UpdatedAt    time.Time  `db:"updated_at" json:"updated_at"`
	CreatedBy    *uuid.UUID `db:"created_by" json:"created_by,omitempty"`
}

type TeamMember struct {
	ID       uuid.UUID  `db:"id" json:"id"`
	TeamID   uuid.UUID  `db:"team_id" json:"team_id"`
	UserID   uuid.UUID  `db:"user_id" json:"user_id"`
	Role     string     `db:"role" json:"role"`
	JoinedAt time.Time  `db:"joined_at" json:"joined_at"`
	LeftAt   *time.Time `db:"left_at" json:"left_at,omitempty"`
	IsActive bool       `db:"is_active" json:"is_active"`

	// Joined user data
	FullName string `db:"full_name" json:"full_name,omitempty"`
	Email    string `db:"email" json:"email,omitempty"`
	Username string `db:"username" json:"username,omitempty"`
}

type Project struct {
	ID               uuid.UUID  `db:"id" json:"id"`
	OrgID            uuid.UUID  `db:"org_id" json:"org_id"`
	Name             string     `db:"name" json:"name"`
	Description      *string    `db:"description" json:"description,omitempty"`
	ProjectManagerID *uuid.UUID `db:"project_manager_id" json:"project_manager_id,omitempty"`
	Status           string     `db:"status" json:"status"`
	Priority         string     `db:"priority" json:"priority"`
	StartDate        *time.Time `db:"start_date" json:"start_date,omitempty"`
	EndDate          *time.Time `db:"end_date" json:"end_date,omitempty"`
	Budget           *float64   `db:"budget" json:"budget,omitempty"`
	Progress         int32      `db:"progress" json:"progress"`
	Metadata         string     `db:"metadata" json:"metadata"`
	CreatedAt        time.Time  `db:"created_at" json:"created_at"`
	UpdatedAt        time.Time  `db:"updated_at" json:"updated_at"`
	CreatedBy        *uuid.UUID `db:"created_by" json:"created_by,omitempty"`
}

type ProjectTeam struct {
	ID         uuid.UUID  `db:"id" json:"id"`
	ProjectID  uuid.UUID  `db:"project_id" json:"project_id"`
	TeamID     uuid.UUID  `db:"team_id" json:"team_id"`
	AssignedAt time.Time  `db:"assigned_at" json:"assigned_at"`
	AssignedBy *uuid.UUID `db:"assigned_by" json:"assigned_by,omitempty"`

	// Joined team data
	TeamName        string `db:"team_name" json:"team_name,omitempty"`
	TeamMemberCount int32  `db:"team_member_count" json:"team_member_count,omitempty"`
}

type ProjectMember struct {
	ID                   uuid.UUID  `db:"id" json:"id"`
	ProjectID            uuid.UUID  `db:"project_id" json:"project_id"`
	UserID               uuid.UUID  `db:"user_id" json:"user_id"`
	Role                 string     `db:"role" json:"role"`
	AllocationPercentage int32      `db:"allocation_percentage" json:"allocation_percentage"`
	JoinedAt             time.Time  `db:"joined_at" json:"joined_at"`
	LeftAt               *time.Time `db:"left_at" json:"left_at,omitempty"`
	IsActive             bool       `db:"is_active" json:"is_active"`

	// Joined user data
	FullName string `db:"full_name" json:"full_name,omitempty"`
	Email    string `db:"email" json:"email,omitempty"`
	Username string `db:"username" json:"username,omitempty"`
}

type Group struct {
	ID          uuid.UUID  `db:"id" json:"id"`
	OrgID       uuid.UUID  `db:"org_id" json:"org_id"`
	Name        string     `db:"name" json:"name"`
	Description *string    `db:"description" json:"description,omitempty"`
	GroupType   string     `db:"group_type" json:"group_type"`
	OwnerID     *uuid.UUID `db:"owner_id" json:"owner_id,omitempty"`
	Status      string     `db:"status" json:"status"`
	Metadata    string     `db:"metadata" json:"metadata"`
	CreatedAt   time.Time  `db:"created_at" json:"created_at"`
	UpdatedAt   time.Time  `db:"updated_at" json:"updated_at"`
	CreatedBy   *uuid.UUID `db:"created_by" json:"created_by,omitempty"`
}

type GroupMember struct {
	ID       uuid.UUID `db:"id" json:"id"`
	GroupID  uuid.UUID `db:"group_id" json:"group_id"`
	UserID   uuid.UUID `db:"user_id" json:"user_id"`
	Role     string    `db:"role" json:"role"`
	JoinedAt time.Time `db:"joined_at" json:"joined_at"`
	IsActive bool      `db:"is_active" json:"is_active"`

	// Joined user data
	FullName string `db:"full_name" json:"full_name,omitempty"`
	Email    string `db:"email" json:"email,omitempty"`
	Username string `db:"username" json:"username,omitempty"`
	TeamName string `db:"team_name" json:"team_name,omitempty"`
}

type Workspace struct {
	ID            uuid.UUID  `db:"id" json:"id"`
	OrgID         uuid.UUID  `db:"org_id" json:"org_id"`
	Name          string     `db:"name" json:"name"`
	Description   *string    `db:"description" json:"description,omitempty"`
	WorkspaceType string     `db:"workspace_type" json:"workspace_type"`
	TeamID        *uuid.UUID `db:"team_id" json:"team_id,omitempty"`
	ProjectID     *uuid.UUID `db:"project_id" json:"project_id,omitempty"`
	OwnerID       *uuid.UUID `db:"owner_id" json:"owner_id,omitempty"`
	Settings      string     `db:"settings" json:"settings"`
	IsPrivate     bool       `db:"is_private" json:"is_private"`
	CreatedAt     time.Time  `db:"created_at" json:"created_at"`
	UpdatedAt     time.Time  `db:"updated_at" json:"updated_at"`
}
